Standalone MPAS-Ocean builds broken

...presumably by #5583

When I try to build standalone MPAS-Ocean on Chrysalis with Intel, OpenMPI, and OpenMP (see below for details), I see:
```
src/libdycore.a(mpas_ocn_time_integration_si.o): In function `ocn_time_integration_si_mp_si_precond_':
mpas_ocn_time_integration_si.F:(.text+0x1d7c9): undefined reference to `dspmv_'
make[1]: *** [Makefile:1032: mpas] Error 1
make[1]: Leaving directory '/gpfs/fs1/home/ac.xylar/compass/add_prescribed_melt_fluxes/e3sm_master/components/mpas-ocean'
make: *** [Makefile:174: ifort] Error 2
```

This was not seen until recently (not in 0273cfad9d when #5343 was merged, the current location of the E3SM-Project submodule in compass).

--- a/components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_si.F
+++ b/components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_si.F
@@ -4324,7 +4324,7 @@ module ocn_time_integration_si
 #endif
 
 !#######################################################################
-#else
+#elif defined(USE_LAPACK)
 
          ! LAPACK on CPU 
          !    (also if no MAGMA & CUBLAS for GPU ; Data staging)
@@ -4332,7 +4332,6 @@ module ocn_time_integration_si
          call DSPMV('U', nPrecVec, 1.0_RKIND, prec_ivmat(:,1), &
                          SIvec_in(1:nPrecVec) , 1, 0.0_RKIND,  &
                          SIvec_out(1:nPrecVec), 1)
-
 #endif
 
       elseif ( trim(config_btr_si_preconditioner) == 'jacobi' ) then

